backend/src/main/java/com/kk/heapoverflow/service/impl/QuestionServiceImpl.java

```diff
@@ -1,32 +1,82 @@
 package com.kk.heapoverflow.service.impl;
 
-import com.kk.heapoverflow.dto.question.QuestionResponseDto;
-import com.kk.heapoverflow.mapper.QuestionMapper;
+import com.kk.heapoverflow.dto.question.response.QuestionAuthorMetadataDto;
+import com.kk.heapoverflow.dto.question.response.QuestionMetadataDto;
+import com.kk.heapoverflow.dto.question.response.QuestionPreviewDto;
+import com.kk.heapoverflow.dto.question.response.QuestionPreviewPageResponseDto;
+import com.kk.heapoverflow.dto.question.response.QuestionTagDto;
 import com.kk.heapoverflow.model.Question;
+import com.kk.heapoverflow.model.User;
+import com.kk.heapoverflow.repostitory.AnswerRepository;
 import com.kk.heapoverflow.repostitory.QuestionRepository;
 import com.kk.heapoverflow.service.QuestionService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
 public class QuestionServiceImpl implements QuestionService {
     private final QuestionRepository questionRepository;
-    private final QuestionMapper questionMapper;
-
-    @Override
-    public Page<QuestionResponseDto> getAllQuestions(Pageable pageable) {
-        return questionRepository.findAll(pageable)
-                .map(questionMapper::toDto);
-    }
+    private final AnswerRepository answerRepository;
 
     @Override
     public Question getQuestionById(Long questionId) {
         Optional<Question> questionOptional = questionRepository.findById(questionId);
         return questionOptional.orElse(null);
     }
+
+    @Override
+    public QuestionPreviewPageResponseDto getQuestionsPreviewPage(Pageable pageable) {
+
+        Page<Question> questionPage = questionRepository.findAll(pageable);
+        List<QuestionPreviewDto> questionPreviews = questionPage.getContent().stream()
+                .map(question -> {
+                    QuestionPreviewDto previewDto = new QuestionPreviewDto();
+                    previewDto.setId(question.getId());
+                    previewDto.setTitle(question.getTitle());
+                    previewDto.setContentShort(question.getContent().substring(0, Math.min(question.getContent().length(), 100))); //TODO 100
+                    previewDto.setAskedAt(question.getCreatedAt());
+                    previewDto.setTags(question.getTags().stream().map(tag -> new QuestionTagDto(tag.getName())).toArray(QuestionTagDto[]::new));
+                    previewDto.setAuthor(mapToAuthorMetadataDto(question.getAuthor()));
+                    previewDto.setMetadata(mapToMetadataDto(question));
+                    return previewDto;
+                })
```

Comment:
refactor to the method
```suggestion
                .map(question -> mapToQuestionPreviewDto(question))
```